include/mysqlpp/udf_wrappers.hpp

```diff
@@ -106,19 +80,83 @@ class generic_udf_base {
       if (e.has_error_code()) {
         auto error_code = e.get_error_code();
         if (error_code == ER_QUERY_INTERRUPTED)
-          my_error(error_code, MYF(0));
+          (*error_reporter)(error_code, MYF(0));
         else
-          my_error(error_code, MYF(0), get_function_label(buffer), e.what());
+          (*error_reporter)(error_code, MYF(0),
+                            get_function_label(buffer, meta_name, item_result),
+                            e.what());
       }
     } catch (const std::exception &e) {
-      my_error(ER_UDF_ERROR, MYF(0), get_function_label(buffer), e.what());
+      (*error_reporter)(ER_UDF_ERROR, MYF(0),
+                        get_function_label(buffer, meta_name, item_result),
+                        e.what());
+    } catch (...) {
+      (*error_reporter)(ER_UDF_ERROR, MYF(0),
+                        get_function_label(buffer, meta_name, item_result),
+                        "unexpected exception");
+    }
+  }
+
+ protected:
+  template <typename ImplType>
+  static void handle_exception() noexcept {
+    using meta_info = udf_impl_meta_info<ImplType>;
+    handle_exception(meta_info::name, meta_info::item_result);
+  }
+  static void handle_init_exception(char *message,
+                                    std::size_t message_size) noexcept {
+    try {
+      // The following suppression is needed exclusively for Clang 5.0 that
+      // has a bug in noexcept specification diagnostics
+#if defined(__clang__) && (__clang_major__ == 5)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wexceptions"
+#endif
+      throw;
```

Comment:
Hmm... It took me quite some time to figure out what happens here. Perhaps it is worth to add comment explaining this, something like: Re-throw exception which is being processed in the caller at the moment.
?

Reply:
Added a comment